### PR TITLE
add tensorflow dependency

### LINKED ISSUE #19: Add Tensorflow 3rdparty dependency
1. Install [Tensorflow](https://www.tensorflow.org/install)
2. Use cmake to find Tensorflow installed on the system (no need to build from scratch)
3. Install Tensorflow library on CI
4. Find and run AlexNet sample that uses Tensorflow

### PR DESCRIPTION
**Pull request by Semyon1104**
fix #19

### CODE REVIEW

File: include/alexnet/alexnet.hpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+#include <tensorflow/c/c_api.h>
+
+#include <iostream>
```
Comment (aobolensk) [inline include/alexnet/alexnet.hpp:4]: Move them to cpp file

File: include/alexnet/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Укажем путь к заголовкам вашей библиотеки
+set(INCLUDE_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/alexnet.hpp")
+
+# Укажем исходные файлы вашей библиотеки
+set(SRC_FILES "${CMAKE_SOURCE_DIR}/src/alexnet/alexnet_src.cpp")
+
+# Создадим цель для вашей библиотеки
+add_library(AlexNetLib STATIC ${INCLUDE_HEADERS} ${SRC_FILES})
+
+set(TENSORFLOW_ROOT "/home/osboxes/.local/lib/python3.6/site-packages/tensorflow")
```
Comment (aobolensk) [inline include/alexnet/CMakeLists.txt:10]: This path is hardcoded and will not work on other machines

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: include/alexnet/alexnet.hpp (hunk 1 above)
Comment (Semyon1104) [reply include/alexnet/alexnet.hpp:4]: remove the hpp file?

**Comment by allnes** [comment]
@Semyon1104 please resolve conflicts

**Review by allnes — APPROVED** [review]

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -29,3 +34,11 @@ file(DOWNLOAD
...
 )
 add_definitions(-DIMAGE_PATH="${CMAKE_CURRENT_BINARY_DIR}/image.jpg")
+file(DOWNLOAD
+    "https://tinyurl.com/2w6eh53v"
+    "${CMAKE_CURRENT_BINARY_DIR}/squeeze.tflite"
+    SHOW_PROGRESS
+    STATUS status_code
+    LOG log_file
+)
+add_definitions(-DMODEL_PATH="${CMAKE_CURRENT_BINARY_DIR}/squeeze.tflite")
```
Comment (aobolensk) [inline test/CMakeLists.txt:44]:
```suggestion
add_definitions(-DMODEL_PATH="${CMAKE_CURRENT_BINARY_DIR}/squeeze.tflite")

```

File: test/model_read/model_read.cpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  file.seekg(0, std::ios::end);
+  size_t file_size = file.tellg();
+  file.seekg(0, std::ios::beg);
+
+  std::vector<char> buffer(file_size);
+
+  ASSERT_NO_THROW(file.read(buffer.data(), file_size););
+  file.close();
+}
```
Comment (aobolensk) [inline test/model_read/model_read.cpp:28]:
```suggestion
}

```

File: include/alexnet/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+endif()
+if(WIN32)
+    set(TensorFlow_INCLUDE_DIRS "${CMAKE_SOURCE_DIR}/3rdparty/tensorflow/include")
+    set(TensorFlow_LIBRARIES "${CMAKE_SOURCE_DIR}/3rdparty/tensorflow/lib/tensorflow.lib")
+
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(tensorflow DEFAULT_MSG TensorFlow_INCLUDE_DIRS TensorFlow_LIBRARIES)
+
+    #message("TensorFlow_INCLUDE_DIRS: ${TensorFlow_INCLUDE_DIRS}")
+    #message("TensorFlow_LIBRARIES: ${TensorFlow_LIBRARIES}")
```
Comment (aobolensk) [inline include/alexnet/CMakeLists.txt:40]:
```suggestion
```

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

**Pull request merged**